sphinx/domains/changeset.py

```diff
@@ -123,7 +123,7 @@ def changesets(self) -> dict[str, list[ChangeSet]]:
     def note_changeset(self, node: addnodes.versionmodified) -> None:
         version = node['version']
         module = self.env.ref_context.get('py:module')
-        objname = self.env.temp_data.get('object', '')
+        objname = self.env.current_document.obj_desc_name
```

Comment:
Now this can be None instead of ''. Wouldn't it still be acceptable as an input to the ChangeSet object constructed just after?